src/main/java/com/google/api/codegen/viewmodel/ReadFileInitCodeLineView.java

```diff
@@ -0,0 +1,47 @@
+/* Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.viewmodel;
+
+import com.google.api.codegen.metacode.InitCodeLineType;
+import com.google.auto.value.AutoValue;
+
+@AutoValue
+public abstract class ReadFileInitCodeLineView implements InitCodeLineView {
+  public InitCodeLineType lineType() {
+    return InitCodeLineType.ReadFileInitLine;
+  }
+
+  public abstract String typeName();
```

Comment:
Isn't the type name already known to be `ByteString` etc?

Reply:
The protobuf type is definitely `bytes`, but the type name may be different in different languages?

Reply:
Oh I think I see what you mean... are you saying that we could render the correct type in snippets?